### PR TITLE
fix: `page` not get reset after filter changes

### LINKED ISSUE #6853: Pagination issue when using filters - pages do not get resseted
### What happened?

If i go to page 4 of the mint filter, disable it and then try another filter, it shows nothing because its trying to get page 4 of that other filter.

https://github.com/kodadot/nft-gallery/assets/36627808/1d25b53c-7881-40bc-8127-41d999fac66c

reference: https://github.com/kodadot/nft-gallery/pull/6745#issuecomment-1682401379

### PR DESCRIPTION
**Pull request by newraina**
**Thank you for your contribution** to the [KodaDot - One Stop Shop for Polkadot NFTs](https://kodadot.xyz).

👇 __ Let's make a quick check before the contribution.

## PR Type

- [x] Bugfix
- [ ] Feature
- [ ] Refactoring

## Context

- [x] Closes #6853 
- [ ] Requires deployment <snek/rubick/worker>

**Why use `useRoute`?**
The `$route` provided in `useNuxtApp()` has a bug where it doesn't get updated when the page parameter in the URL is updated (https://github.com/nuxt/nuxt/issues/13032), replacing it with `useRoute` fixes this.

#### Before submitting pull request, please make sure:

- [x] My contribution builds **clean without any errors or warnings**
- [x] I've merged recent default branch -- **main** and I've no conflicts
- [x] I've tried to respect high code quality standards
- [x] I've didn't break any original functionality

#### Optional

- [ ] I've tested it at </ksm/collection>
- [ ] I've tested PR on mobile
- [ ] I've written unit tests 🧪
- [ ] I've found edge cases

#### Did your issue had any of the "$" label on it?

- [x] Fill up your DOT address: [Payout](https://canary.kodadot.xyz/dot/transfer?target=1CAv6Zq3yVxL3eKhC94GWTWVwp1w4jZbqeZ6wXx1rPAhrce&usdamount=0&donation=true)

#### Community participation

- [x] [Are you at KodaDot Ecosystem Telegram?](https://t.me/kodadot_eco)

## Screenshot 📸

- [ ] My fix has changed **something** on UI; a screenshot is best to understand changes for others.

### <samp>🤖 Generated by Copilot at 77548c9</samp>

Refactored `History.vue` to use Vue Router 4 and pagination. This improves the performance and usability of the profile history tab.

### <samp>🤖 Generated by Copilot at 77548c9</samp>

> _`History.vue` changed_
> _Vue Router 4 and pagination_
> _Autumn leaves fall fast_

### CODE REVIEW

File: components/profile/activityTab/History.vue [hunk 1]
```diff
@@ -90,7 +90,7 @@ const prop = withDefaults(
 )
 const emit = defineEmits(['setPriceChartData'])
 
-const { $route } = useNuxtApp()
+const $route = useRoute()
```
Comment (roiLeo) [inline components/profile/activityTab/History.vue:93]:
```suggestion
const route = useRoute()
```

File: components/profile/activityTab/History.vue [hunk 2]
```diff
@@ -258,4 +258,11 @@ const createTable = (): void => {
 watch(() => prop.events, createTable)
 
 watch(event, updateDataByEvent)
+
+watch(
+  () => $route.query?.page,
```
Comment (roiLeo) [inline components/profile/activityTab/History.vue:263]:
```suggestion
  () => route.query?.page,
```

**Review by roiLeo — APPROVED** [review]

**Comment by yangwao** [comment]
pay 50 usd

**Comment by yangwao** [comment]
😍 Perfect, I’ve sent the payout 
💵 $50 @ 4.24 USD/DOT ~ 11.792 $DOT
🧗 [1CAv6Zq3yVxL3eKhC94GWTWVwp1w4jZbqeZ6wXx1rPAhrce](https://polkadot.subscan.io/account/1CAv6Zq3yVxL3eKhC94GWTWVwp1w4jZbqeZ6wXx1rPAhrce)
🔗 [0xc86b3de30f9141f1afba407a2337894a3e36b762a3fe17e55f74ae0786f62fa1](https://polkadot.subscan.io/extrinsic/0xc86b3de30f9141f1afba407a2337894a3e36b762a3fe17e55f74ae0786f62fa1)

🪅 Let’s grab another issue and get rewarded! 
🪄 [github.com/kodadot/nft-gallery/issues](https://github.com/kodadot/nft-gallery/issues/)

**Pull request merged**